File: 0098-validate-binary-search-tree/0098-validate-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void inorder(TreeNode root,List<Integer> list){
        if(root==null){
            return;
        }
        inorder(root.left,list);
        list.add(root.val);
        inorder(root.right,list);
    }
    public boolean isValidBST(TreeNode root) {
        List<Integer> list=new ArrayList<>();
        inorder(root,list);
        int cnt=0;
        for(int i=1;i<list.size();i++){
            if(list.get(i)<=list.get(i-1)){
                cnt++;
            }
        }
        return cnt<1;
    }
}

File: 0098-validate-binary-search-tree/0098-validate-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isValid(TreeNode* root,long mn,long mx){
        if(root==NULL){
            return true;
        }
        if(root->val<=mn || root->val>=mx){
            return false;
        }
        return isValid(root->left,mn,root->val) && isValid(root->right,root->val,mx);
    }
    bool isValidBST(TreeNode* root) {
        
        return isValid(root,LONG_MIN,LONG_MAX);
    }
};